src/main/java/com/google/sps/servlets/GetVendorServlet.java

```diff
@@ -0,0 +1,105 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.HttpServletUtils;
+import com.google.sps.data.Vendor;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Retrieve all the vendor information, including saleCard and profile picture
+ * GET: For clients wanting info of a specific vendor
+ * POST: For vendors wanting to see their info
+ */
+@WebServlet("/get-vendor")
+public class GetVendorServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String vendorId = HttpServletUtils.getParameter(request, "vendorId", "");
+    float latitude, longitude;
+    GeoPt clientLocation;
+    try {
+      // If not provided, we set them to 360 to throw an error when trying to use them to create a GeoPt
+      latitude = Float.parseFloat(HttpServletUtils.getParameter(request, "lat", "360"));
+      longitude = Float.parseFloat(HttpServletUtils.getParameter(request, "long", "360"));
+      clientLocation = new GeoPt(latitude, longitude);
+    } catch (NumberFormatException e) {
+      System.out.println("The string is not a parsable float: " + e);
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Wrong format of parameters");
+      return;
+    } catch (IllegalArgumentException e) {
+      System.out.println(e);
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Latitude and/or longitude outside range");
+      return;
+    }
+
+    if (vendorId.isEmpty()) {
+      System.out.println("No vendor provided");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No vendor provided.");
+      return;
+    }
+
+    Entity vendorEntity = HttpServletUtils.getVendorEntity(vendorId);
+    if (vendorEntity == null) {
+      System.out.println("Vendor Account does not exist");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Vendor Account does not exist");
+      return;
+    }
+
+    Vendor vendorObject = new Vendor(vendorEntity);
+    // Add the distance between client and vendor's business
+    if (vendorObject.getSaleCard() != null) {
+      float distanceClientVendor = HttpServletUtils.computeGeoDistance(
+          clientLocation, vendorObject.getSaleCard().getLocation().getSalePoint());
+      vendorObject.getSaleCard().setDistanceFromClient(distanceClientVendor);
+    }
+
+    response.setContentType("application/json;");
+    response.setCharacterEncoding("UTF-8");
+    response.getWriter().println(new Gson().toJson(vendorObject));
+  }
+
+  @Override
```

Comment:
Could you explain with a comment why do we need to have POST and GET methods that returns almost the same vendor entity?

Reply:
You mean like _implementation comments_ `// Comment...`? 
It is explained a bit at the beginning of the servlet, but sure I can add comments before each method too.